Example using Vault to fetch wnd password (#3250)
owncloud/docs: modules/admin_manual/pages/enterprise/external_storage/windows-network-drive_configuration.adoc

password in plain text somehow. If not, additional operations might be required to get the password as
plain text and inject it in the listener.

Example using "Vault" as secrets store

> * See https://learn.hashicorp.com/collections/vault/getting-started

Reviewer: can you make this link a macro as we do with the other links?
see on top of the page
like: `:vault-getting-started-url: https://learn.hashicorp.com/collections/vault/getting-started`
and here: `See the {vault-getting-started-url}[get started here guide]

on how to setup Vault as your secrets store. Then use something like the following

> [source,console,subs="attributes+"]

Reviewer 2: What does that mean?

Author: This is adoc markup

Reviewer 2: I now I get it.. :sweat_smile:

----

> vault kv get -field=password secret/samba | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-

Reviewer 2: Can we change the flags of the wnd app? `--password-file` looks wrong.

Reviewer 2: Of course this needs to be done in the app not the docs. :)

----

> Use Vault's ACLs to limit access of the token. Destroy the token after starting the service

Reviewer 2: > Destroy the token after starting the service during boot with systemd.

What do you mean by that?

Author: Vault requires an access token, so theoretically one could provide it at boot-time to fetch the secret and then remove the token after the secret was fetched. The secret would reside in memory of the running wnd process only.  Not sure about this one either? Maybe too convoluted?

Reviewer 2: But which token do you mean?
The wnd secret? That should stay in the vault, right?
Or the access token to the vault? But how would it be provided? Manually by a person or is this an automated step?
I probably know too little about the vault setup process... :see_no_evil:

Author: Ah right, I rewword

=== 3rd Party Software Examples

Third party password managers or processes can be integrated. The only requirement is that they have to provide the password in plain text somehow. If not, additional operations might be required to get the password as plain text and inject it in the listener.

==== plainpass

This provides a bit more security because the `/tmp/plainpass` password as shown below should be owned by root and only root should be able to read the file (0400 permissions); Apache, particularly, shouldn't be able to read it. It's expected that root will be the one to run this command.

[source,console,subs="attributes+"]
----
cat /tmp/plainpass | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
----

==== base64

> Similar to plainpass, the contents in this case gets encoded in the {base64-url}[Base64 format]. There's not much security, but it has additional obfuscation.

Reviewer 3: ```suggestion
Similar to plainpass, the content in this case gets encoded in the {base64-url}[Base64 format]. There's not much security, but it has additional obfuscation.
```

[source,console,subs="attributes+"]
----
base64 -d /tmp/encodedpass | \
   {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
----

==== pass

Example using "pass"

> * You can go through {pass-url}[manage passwords from the command line] to setup the keyring for whoever will fetch the password (probably root) and then use something like the following

Reviewer 3: ```suggestion
* You can go through {pass-url}[manage passwords from the command line] to set up the keyring for whoever will fetch the password (probably root) and then use something like the following:
```

[source,console,subs="attributes+"]
----
pass the-password-name | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
----

==== HashiCorp Vault

> This example uses {vaultproject-url}[Vault] as secrets store. See {hashicorp-url}[HCP Vault] on how to setup the secrets store. Then use something like the following:

Reviewer 4: ```suggestion
This example uses {vaultproject-url}[Vault] as the secrets store. See {hashicorp-url}[HCP Vault] on how to setup the secrets store. Then use something like the following:
```

[source,console,subs="attributes+"]
----
vault kv get -field=password secret/samba | {occ-command-example-prefix} wnd:listen <host> <share> <username> --password-file=-
----

> Use Vault's ACLs to limit access of the token. Destroy the token after starting the service during boot with systemd.

Reviewer 4: ```suggestion
Use Vault's ACLs to limit access to the token. Destroy the token after starting the service during boot with systemd.
```
